### DISCUSSION TITLE
Halide GPU example fails

### DISCUSSION THREAD

**Discussion by harishch4** [halide/Halide#5434] [Q&A] [answered]
I'm new to Halide and trying the run the provided tutorials. When running GPU  example([lesson_12_using_the_gpu][1]), i am getting the below error.

    harry@harry-GL553VD:~/Halide/build/tutorial$ ./lesson_12_using_the_gpu 
    Running pipeline on CPU:
    Running pipeline on GPU:
    : CommandLine Error: Option 'help-list' registered more than once!
    LLVM ERROR: inconsistency in registered CommandLine options

I'm running this on a Linux machine with NVIDIA GPU.

        harry@harry-GL553VD:~/Halide/build/tutorial$ clinfo
    Number of platforms                               1
      Platform Name                                   NVIDIA CUDA
      Platform Vendor                                 NVIDIA Corporation
      Platform Version                                OpenCL 1.2 CUDA 11.0.185
      Platform Profile                                FULL_PROFILE
      Platform Extensions                             cl_khr_global_int32_base_atomics cl_khr_global_int32_extended_atomics cl_khr_local_int32_base_atomics cl_khr_local_int32_extended_atomics cl_khr_fp64 cl_khr_byte_addressable_store cl_khr_icd cl_khr_gl_sharing cl_nv_compiler_options cl_nv_device_attribute_query cl_nv_pragma_unroll cl_nv_copy_opts cl_nv_create_buffer cl_khr_int64_base_atomics cl_khr_int64_extended_atomics
      Platform Extensions function suffix             NV
    
      Platform Name                                   NVIDIA CUDA
    Number of devices                                 1
      Device Name                                     GeForce GTX 1050
      Device Vendor                                   NVIDIA Corporation
      Device Vendor ID                                0x10de
      Device Version                                  OpenCL 1.2 CUDA
      Driver Version                                  450.36.06
      Device OpenCL C Version                         OpenCL C 1.2 

  [1]: https://github.com/halide/Halide/blob/master/tutorial/lesson_12_using_the_gpu.cpp

**Comment by alexreinking** [comment, marked as answer]
It looks like you built Halide and maybe LLVM yourself and potentially incorrectly... can you post the commands you used to build each one?

**Reply by harishch4** [reply]
Hi Alex,
I followed the below steps.
[https://stackoverflow.com/questions/64588796/when-trying-to-build-halide-on-linux-system-with-llvm-10-0-on-release-10-x-and-m/64605846#64605846](https://stackoverflow.com/questions/64588796/when-trying-to-build-halide-on-linux-system-with-llvm-10-0-on-release-10-x-and-m/64605846#64605846)

**Reply by alexreinking** [reply]
That still doesn't show how you built LLVM.

**Reply by harishch4** [reply]
I'm using the LLVM-10.0.0 pre-built binaries obtained form [https://releases.llvm.org/download.html#10.0.0](https://releases.llvm.org/download.html#10.0.0)
